managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImpl.java

```diff
@@ -398,10 +398,11 @@ private static ManagedLedgerInfo updateMLInfoTimestamp(ManagedLedgerInfo info) {
     }
 
     private static MetaStoreException getException(Throwable t) {
-        if (t.getCause() instanceof MetadataStoreException.BadVersionException) {
-            return new ManagedLedgerException.BadVersionException(t.getMessage());
+        Throwable actEx = FutureUtil.unwrapCompletionException(t);
+        if (actEx instanceof MetadataStoreException.BadVersionException) {
+            return new ManagedLedgerException.BadVersionException(actEx.getMessage());
         } else {
-            return new MetaStoreException(t);
+            return new MetaStoreException(actEx);
```

Comment:
if `actEx` is already a `MetaStoreException`, couldn't we return it without wrapping?

Reply:
Improved